ci/docker/datafusion_consumer_functions.Dockerfile

```diff
@@ -3,7 +3,7 @@ FROM ubuntu:22.04
 ENV PYTHONUNBUFFERED=1
 RUN apt-get update && apt-get install -y python3.10 && ln -sf python3 /usr/bin/python
 RUN apt install -y pip
-RUN pip install --upgrade pip setuptools pytest pytest-snapshot substrait pyarrow protobuf duckdb filelock datafusion==40.1.0 ibis_substrait JPype1 substrait-validator
+RUN pip install --upgrade pip setuptools pytest pytest-snapshot substrait pyarrow protobuf duckdb filelock datafusion==41.0.0 ibis_substrait JPype1 substrait-validator
```

Comment:
We're referring to this same version in at least a dozen places.  Perhaps we could put this in a file of constant version dependencies so we don't miss any locations?